lib/slug.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+# Helper class to generate slugs out of names or titles.
+class Slug
+  CONTROL_CHARACTERS_RE = /[\x00-\x1F\x7F]/
+
+  def self.generate(string)
+    raise(
+      ArgumentError,
+      "Can't generate a slug from a nil value."
+    ) if string.nil?
+
+    string.unicode_normalize.downcase(:fold).
```

Comment:
Is there a reason here to use custom logic vs. rely on https://apidock.com/rails/v6.0.0/ActiveSupport/Inflector/parameterize ?

Reply:
See comment below.